### ISSUE TITLE
question about guided backprop

### ISSUE THREAD

**Issue by narrowsnap** [utkuozbulak/pytorch-cnn-visualizations#65]
Hi, thank you for this great repo.
I'm confused when I'm reading guided_backprop.py.
On line 70-73, the gradient of output is [0, 0, ..., 1, 0]. 
```
one_hot_output = torch.FloatTensor(1, model_output.size()[-1]).zero_()
one_hot_output[0][target_class] = 1
# Backward pass
model_output.backward(gradient=one_hot_output)
```

I changed target class while draw cat_dog_Guided_BP_color map. But I find no matter what class is set, it always looks the same.

For example:
cat_dog_Guided_BP_color_target_class_10.jpg
![cat_dog_Guided_BP_color_target_class_10](https://user-images.githubusercontent.com/17559987/71573549-6c596580-2b1f-11ea-9ac0-4fdab08a19d2.jpg)

cat_dog_Guided_BP_color_target_class_100.jpg
![cat_dog_Guided_BP_color_target_class_100](https://user-images.githubusercontent.com/17559987/71573378-89416900-2b1e-11ea-94f1-85c212590167.jpg)

cat_dog_Guided_BP_color_target_class_243.jpg(ground truth)
![cat_dog_Guided_BP_color_target_class_243](https://user-images.githubusercontent.com/17559987/71573388-93636780-2b1e-11ea-90d4-f72839dc6ea4.jpg)

cat_dog_Guided_BP_color_target_class_500.jpg
![cat_dog_Guided_BP_color_target_class_500](https://user-images.githubusercontent.com/17559987/71573395-9bbba280-2b1e-11ea-8feb-ac35ea3f76b2.jpg)

cat_dog_Guided_BP_color_target_class_890.jpg
![cat_dog_Guided_BP_color_target_class_890](https://user-images.githubusercontent.com/17559987/71573407-a4ac7400-2b1e-11ea-9c55-cffdfa2bc8ed.jpg)

**Comment by Bear-kai** [comment]
They just look similar, not the same. This happen in not only guided backprop, but also other vis technique. Actually, it is reasonable~

**Comment by narrowsnap** [comment]
@Bear-Kai Dose this mean network recognizes objects via edge? 
And I don't understand why set gradient=one_hot_output rather than set gradient=ones_like(model_output).  Any explain will be thanked.

**Comment by Bear-kai** [comment]
@narrowsnap 
Since you only care about the logit (and the gradient from the logit) corresponding to the GT. one_hot_output is considered as a mask to filter out all non-GT logits.

**Comment by narrowsnap** [comment]
@Bear-kai Thank you! Get it.

**Comment by utkuozbulak** [comment]
Thank you for the answer @Bear-kai  !

**Comment by narrowsnap** [comment]
@Bear-kai @utkuozbulak
